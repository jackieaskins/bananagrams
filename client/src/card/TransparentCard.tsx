import { CSSObject } from '@emotion/react';
import { Card } from 'antd';
import { CardProps } from 'antd/lib/card';

interface TransparentCardProps extends CardProps {
  cardCSS?: CSSObject;
}

const TransparentCard = ({
  cardCSS = {},
  children,
  ...rest
}: TransparentCardProps): JSX.Element => (
  <Card
    {...rest}
    css={{ backgroundColor: 'rgba(255, 255, 255, 0)', ...cardCSS }}
  >
    {children}
  </Card>
);

export default TransparentCard;
