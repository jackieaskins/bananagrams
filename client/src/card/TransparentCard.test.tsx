import { shallow } from 'enzyme';

import TransparentCard from './TransparentCard';

describe('<TransparentCard />', () => {
  const renderComponent = (propOverrides = {}) =>
    shallow(<TransparentCard {...propOverrides} />);

  it('card with transparent background', () => {
    expect(renderComponent()).toMatchSnapshot();
  });
});
