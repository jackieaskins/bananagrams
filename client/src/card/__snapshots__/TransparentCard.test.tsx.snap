// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransparentCard /> card with transparent background 1`] = `
.emotion-01lbzn79-TransparentCard {
  background-color: rgba(255, 255, 255, 0);
}

<Card
  className="emotion-0"
/>
`;
